packages/react/src/__tests__/state.test.tsx

```diff
@@ -271,3 +272,83 @@ it('should modify state after receiving response from the server', async () => {
     </DocumentFragment>
   `)
 })
+
+it('should inject state (initial value) into function handler', async () => {
+  const onTemplateEvent = jest.fn().mockReturnValue(response(null))
+  const isChecked = state(false)
+  const dataSource: DataSource = {
+    get: () => ({
+      subscribe: () => jest.fn(),
+      get() {
+        return [
+          {
+            $: 'component',
+            key: 'button',
+            component: 'View',
+            props: {
+              ['data-testid']: 'button',
+              onClick: {
+                $: 'event',
+                handler: {
+                  $: 'handler',
+                  state: { isChecked },
+                },
+              },
+            },
+          },
+        ]
+      },
+    }),
+    sendEvent: onTemplateEvent,
+  }
+  const component = render(<Template components={BUILT_IN_COMPONENTS} dataSource={dataSource} />)
+  await act(async () => {
+    fireEvent.click(component.getByTestId('button'))
+  })
+  expect(onTemplateEvent.mock.lastCall[0].payload).toEqual({ isChecked: false })
+})
+
+it('should inject current state value into function handler', async () => {
+  const onTemplateEvent = jest.fn().mockReturnValue(response(null))
+  const isChecked = state(false)
+  const dataSource: DataSource = {
+    get: () => ({
+      subscribe: () => jest.fn(),
+      get() {
+        return [
+          {
+            $: 'component',
+            key: 'button',
+            component: 'View',
+            props: {
+              ['data-testid']: 'button',
+              onClick: {
+                $: 'event',
+                handler: {
+                  $: 'handler',
+                  state: { isChecked },
+                },
+              },
+            },
+          },
+          {
+            $: 'component',
+            key: 'checkbox',
+            component: 'View',
+            props: {
+              ['data-testid']: 'checkbox',
+              onClick: setStateAction(isChecked, true),
+            },
+          },
+        ]
+      },
+    }),
+    sendEvent: onTemplateEvent,
+  }
+  const component = render(<Template components={BUILT_IN_COMPONENTS} dataSource={dataSource} />)
+  await act(async () => {
+    fireEvent.click(component.getByTestId('checkbox'))
+    fireEvent.click(component.getByTestId('button'))
```

Comment:
This is broken test case - need to fix state updates when concurrent event happen in a single tick.

Reply:
This test case is fixed in #76